CONTRIBUTING.md

```diff
@@ -0,0 +1,133 @@
+
+## Bug reports
+
+A bug is a *demonstrable problem* that is caused by code in the repository. Good bug reports are extremely helpful - thank you!
+
+Guidelines for bug reports:
+
+1. **Use the GitHub issue search** - check if the issue has *already* been reported.
+2. **Check if the issue has been fixed** - try to reproduce it using the latest `main` branch in the repository.
+3. **Isolate the problem** - create a reduce test case and a live example.
+
+A good bug report shouldn't leave others needing to chase you up for more information. Please try to be as detailed as possible in your report. What is your environment? What steps will reproduce the issue? What browser(s) and OS experience the problem? What would you expect to be the outcome? All these details will help people to fix any potential bugs.
+
+Example:
+
+> Short and descriptive example bug report title
+>
+> A summary of the issue and the browser/OS environment in which it occurs. If suitable, include the steps required to reproduce the bug.
+>
+> 1. This is the first step
+> 2. This is the second step
+> 3. Further steps, etc.
+> <url> - a link to the reduced test case
+>
+> Any other information you want to share that is relevant to the issue being reported. This might include the lines of code that you have identified as causing the bug, and potential solutions (and your opinions on their merits).
+
+## Feature requests
+
+Feature requests are welcome, but take a moment to find out whether your idea fits with the scope and aims of the project. It's up to *you* to make a strong case to convince the project's developers of the merits of this feature. Please provide as much detail and context as possible.
+
+# Engineering standards
+
+All ongoing development for an upcoming release gets commited to the **`main`** branch. The `main` branch technically serves as the "development" branch as well, but all code that is committed to the `main` branch should be considered *stable*.
+
+Ensure any changes are clear and well-documented. When we say "well-documented":
+
+* If the changes include code, ensure all additional code has documentation in and around it. This includes documenting the definitions of fnuctoins, statements in code, sections.
+* The most helpful code comments explain why, establish context, or efficiency summarize how. Avoid simply repeating details from declarations. When in doubt, favour overexplaining to underexplaining.
+* Code comments should be consistent with their languge conventions. For example, please use [JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) for TypeScript/JavaScript, or [NatSpec](https://docs.soliditylang.org/en/v0.8.13/natspec-format.html) for Solidity.
+* Any new features must have corresponding user documentation. Any removed features must have their user documentation removed from the documents.
+* Do **NOT** submit commented-out code. If the code does not need to be used anymore, please remove it.
```

Comment:
I agree, we have to keep the code clear. Sometimes, it is required to switch off one function. If we just remove this part of the code, when it will be required to restore temporarily removed code other developers simply will not knows about the existence of the removed code and have to rewrite it from scratch.
I guess this requirement should not be mandatory.

Reply:
It's important to view this requirement with respect to the `main` branch. People are free to fork and use their own branches prior to a PR merging into `main` and have commented out code in their feature branch (as a reminder etc). Does this alleviate your concern? (basically, anything in `main` should be clean so others who may come in and contribute - without having to assume what the commented code is there for etc).